<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav)"/>
  <arg name="altitude" default="3.0"/>
  <arg name="x_min" default="-4.0"/>
  <arg name="x_max" default="4.0"/>
  <arg name="y_min" default="-3.0"/>
  <arg name="y_max" default="4.0"/>
  <arg name="z_min" default="2.0"/>
  <arg name="z_max" default="6.0"/>

  <group ns="$(arg uav_name)">
    
    <node name="random_flier" pkg="random_flier" type="random_flier" output="screen" >

      <param name="altitude" type="double" value="$(arg altitude)" />
      <param name="x_min" type="double" value="$(arg x_min)"/>
      <param name="x_max" type="double" value="$(arg x_max)"/>
      <param name="y_min" type="double" value="$(arg y_min)"/>
      <param name="y_max" type="double" value="$(arg y_max)"/>
      <param name="z_min" type="double" value="$(arg z_min)"/>
      <param name="z_max" type="double" value="$(arg z_max)"/>

      <!-- Subscribers -->
      <remap from="~diagnostics_in" to="control_manager/mpc_tracker/diagnostics" />
      <remap from="~odom_in" to="odometry/odom_main" />

      <!-- Publishers -->
      <remap from="~goto_out" to="control_manager/goto" />
    
    </node>
  
  </group>
</launch>
